packages/kbn-unified-data-table/src/components/custom_control_columns/additional_row_control/row_menu_control_column.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment, useCallback, useState } from 'react';
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiDataGridCellValueElementProps,
+  EuiDataGridControlColumn,
+  EuiPopover,
+  EuiScreenReaderOnly,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import { DataTableRowControl, Size } from '../../data_table_row_control';
+import type { RowControlColumn, RowControlProps } from '../../../types';
+import { DEFAULT_CONTROL_COLUMN_WIDTH } from '../../../constants';
+import { useControlColumn } from '../../../hooks/use_control_column';
+
+/**
+ * Menu button under which all other additional row controls would be placed
+ */
+export const RowMenuControlCell = ({
+  rowControlColumns,
+  ...props
+}: EuiDataGridCellValueElementProps & {
+  rowControlColumns: RowControlColumn[];
+}) => {
+  const rowProps = useControlColumn(props);
+  const [isMoreActionsPopoverOpen, setIsMoreActionsPopoverOpen] = useState<boolean>(false);
+
+  const buttonLabel = i18n.translate('unifiedDataTable.grid.additionalRowActions', {
+    defaultMessage: 'Additional actions',
+  });
+
+  const getControlComponent: (id: string) => React.FC<RowControlProps> = useCallback(
+    (id) =>
+      ({ 'data-test-subj': dataTestSubj, color, disabled, label, iconType, onClick }) => {
+        return (
+          <EuiContextMenuItem
+            data-test-subj={dataTestSubj ?? `unifiedDataTable_rowMenu_${id}`}
+            disabled={disabled}
+            icon={iconType}
+            color={color}
+            onClick={() => {
+              onClick?.(rowProps);
+              setIsMoreActionsPopoverOpen(false);
+            }}
+          >
+            {label}
+          </EuiContextMenuItem>
+        );
+      },
+    [rowProps, setIsMoreActionsPopoverOpen]
+  );
+
+  return (
+    <EuiPopover
+      id="rowMenuActionsPopover"
+      button={
+        <DataTableRowControl size={Size.normal}>
+          <EuiToolTip content={buttonLabel} delay="long">
+            <EuiButtonIcon
+              size="xs"
+              iconSize="s"
+              aria-label={buttonLabel}
+              data-test-subj={`unifiedDataTable_${props.columnId}`}
```

Comment:
Nit: should we make this test subj more specific?

Reply:
It unwraps as `unifiedDataTable_additionalRowControl_menuControl`. What information would be useful to include too?